README.md

```diff
@@ -441,4 +443,33 @@ newElement.innerHTML = '<div></div>';
 document.querySelector('main').diffRelease(newElement);
 ```
 
+#### Middleware
+
+A controlled way of exposing the internals of diffHTML to external tooling and
+has a very simple interface for creating and consuming.
+
+Two interesting projects that illustrate use cases for middleware:
+
+- [Logger](https://github.com/tbranyen/diffhtml-logger)
+- [Components](https://github.com/tbranyen/diffhtml-components)
+
+##### Consuming
+
+Very simple, use the `use` method.
+
+``` javascript
+import logger from 'diffhtml-logger';
+import { use } from 'diffhtml';
+
+use(logger);
+```
+
+##### Create
+
+``` javascript
+const logMiddleware = (
+```
```

Comment:
Yup, in fact I'm going to remove all the README changes and just focus on them in the readme-changes branch. Pushing towards a 1.0 I'm just going to merge whatever in master for a bit.